models/group.go

```diff
@@ -212,6 +213,12 @@ func PutGroup(g *Group) error {
 	if err := g.Validate(); err != nil {
 		return err
 	}
+
+	record, err := GetGroupByName(g.Name, g.UserId)
+	if err == nil && record.Id != g.Id {
```

Comment:
Let's remove the checks to this new error you're introducing. I see the reasoning behind it, but it's better to keep these changes focused solely on the public key support you're adding.

Reply:
I believe this issue isnt from my input, what I found during development of this is that put requests can modify the names of landing pages/groups/etc and set them to be the same name as other entries. 
However you are correct in that this should be a seperate merge on its own, rather than being lumped in with the public key stuff.